### ISSUE TITLE
Social links: Prevent empty links and improve UI description

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#60044]
### Description

After users edit and empty a social link 'label', the link on the front end is empty. No text, no aria-label. Just an empty link.

WordPress should always output semantic, SEO friendly, and accessible HTML. No empty links should ever be allowed.

The resulting HTML output may end up being, for example:
```
<a href="https://github.com/myusername" class="wp-block-social-link-anchor"><svg ...></svg><span class="wp-block-social-link-label"></span></a>
```

Rendering empty links on the front end doesn't help produce high quality, SEO friendly and accessible content. This must be avoided.

Users already have the option to visually hide the social link text and only show the icon. As such, there is no reason to make this link text empty.  It must always be populated with some meaningful text.

This issue is even worse when 'Show labels' in the block setttings panel is disabled as the fact the link is empty is not visible to editors. There's no feedback at all the link is empty. 

Technically, this:

https://github.com/WordPress/gutenberg/blob/c434659b89d9319b7f490aa0405803cfcb12957c/packages/block-library/src/social-link/edit.js#L116

doesn't work very well _after_ users edit the link text.
- The `label` prop is initially undefined, so the default link text is the social network name. So far so good.
- After users edit the link text and then empty it, the `label` prop is set to empty string.
- As such, the empty string is assigned to the link text.

### Step-by-step reproduction instructions

- Add a Social Icons block.
- To better understand this bug, enable 'Show labels' in the block settings panel. Screenshot:

![Screenshot 2024-03-20 at 15 50 07](https://github.com/WordPress/gutenberg/assets/1682452/ba552361-807f-4192-854c-40ffc54e9d1b)

- The social icon links will show their default label, which is the social network name.
- Note: the term 'label' is a bit inappropriate here because this is actually the _link text_.
- Click the plus icon in the block placeholder and add a social link, for example GitHub.
- Click the link placeholder to add an actual link.
- Note: the link input field placeholder is `Enter address` which is inconsistent with other terminology used across the editor.
- Enter a link e.g. `https://github.com/myusername` and save.
- In the block settings panel, expand the 'GitHub label' section. 
- Observe the Link label input field is empty. Expected: to be populated with the default link text e.g. `GitHub`. 
- Note: the description `Briefly describe the link to help screen reader users.` isn't appropriate when 'Show labels' is enabled. In this case, this is the _visible_ link text. Not something only for screen readers. Also, we should rarely mention specific assistive technology. This is the actual link text and should be clarified to users.
- Enter something in the input field.
- Observe the social link text in the editor canvas updates on the fly with the value you entered.
- Empty the input field and save.
- Observe the social link text in the editor canvas disappeared. Expected: to be restored to the default social name.
- Go to the front end and inspect the social link.
- Observe the link is an empty link with no text or any aria-label.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Additionally, I noticed the popover 'Apply' button doesn't show a tooltip, not sure why.

![Screenshot 2024-03-20 at 16 47 19](https://github.com/WordPress/gutenberg/assets/1682452/ebebf3b5-285c-464b-b33c-1b3e84dcb0db)

**Comment by ramonjd** [comment]
> After users edit and empty a social link 'label', the link on the front end is empty. No text, no aria-label. Just an empty link.

I also noticed that the frontend will still render `<span class="wp-block-social-link-label screen-reader-text"></span>` after removing the label in the editor.

This PR is a quick win to use the fall back value in these cases so there's at least some text there.

- https://github.com/WordPress/gutenberg/pull/60060

**Comment by afercia** [comment]
> Additionally, I noticed the popover 'Apply' button doesn't show a tooltip, not sure why.


After some debugging, the tooltip doesn't work because the `SocialLinkURLPopover` is nested inside a `Button`, see:

https://github.com/WordPress/gutenberg/blob/f9403411d72c36de037a24b566a08180fe69673a/packages/block-library/src/social-link/edit.js#L160-L182

So that after https://github.com/WordPress/gutenberg/pull/57202 the tooltip will not show. Cc @ciampo while I think this instance can be easily fixed, looks like https://github.com/WordPress/gutenberg/pull/57202 may introduce other regressions still undiscovered. How we can catch and prevent these cases? Thanks.
